qiskit/aqua/algorithms/classifiers/vqc.py

```diff
@@ -469,13 +499,54 @@ def get_optimal_vector(self):
             self._ret['min_vector'] = ret.get_counts(qc)
         return self._ret['min_vector']
 
+    @property
+    def feature_map(self):
+        """Return the feature map."""
+        return self._feature_map
+
+    @feature_map.setter
+    def feature_map(self, feature_map):
+        """Set the feature map.
+
+        Also sets the number of qubits, the internally stored feature map parameters and,
+        if the feature map is a circuit, the order of the parameters.
+        """
+        if isinstance(feature_map, QuantumCircuit):
+            # patch the feature dimension to the circuit
+            feature_map.feature_dimension = len(feature_map.parameters)
+
+            # store the parameters
+            self._num_qubits = feature_map.num_qubits
+            self._feature_map_params = list(feature_map.parameters)
+            self._feature_map = feature_map
+        elif isinstance(feature_map, FeatureMap):
+            self._num_qubits = feature_map.num_qubits
+            self._feature_map_params = ParameterVector('x', length=feature_map.feature_dimension)
+            self._feature_map = feature_map
+        elif feature_map is None:
```

Comment:
This is currently needed because the initializer calls this setter, and there the feature map might be None. How do you think we should handle this? Set a default feature map?